### PR TITLE
Airlock - API - approve/reject a request

### PR DESCRIPTION
**Pull request by anatbal**
# PR fixes issue #1840

## What is being addressed
** testing will be addressed in a subsequential pr**
- Implement the review endpoint which allows the user to approve/reject a request
- Create a new object - an airlock review
- Fixed exception to return 400 whenever validation fails
- Since status is not an attribute of airlock review, renamed it and changed the history mechanism so it will be included only on airlock requests
- Fix minor typos

### CODE REVIEW

File: api_app/api/routes/airlock.py [hunk 1]
```diff
@@ -25,7 +27,7 @@ async def create_draft_request(airlock_request_input: AirlockRequestInCreate, us
     try:
         airlock_request = airlock_request_repo.create_airlock_request_item(airlock_request_input, workspace.id)
     except (ValidationError, ValueError) as e:
-        logging.error(f"Failed create air lock request model instance: {e}")
+        logging.error(f"Failed create airlock request model instance: {e}")
```
Comment (eladiw) [inline api_app/api/routes/airlock.py:30]:
```suggestion
        logging.error(f"Failed creating airlock request model instance: {e}")
```

File: api_app/api/routes/airlock.py [hunk 2]
```diff
@@ -35,3 +37,17 @@ async def create_draft_request(airlock_request_input: AirlockRequestInCreate, us
 async def create_submit_request(airlock_request=Depends(get_airlock_request_by_id_from_path), user=Depends(get_current_workspace_owner_or_researcher_user), airlock_request_repo=Depends(get_repository(AirlockRequestRepository))) -> AirlockRequestInResponse:
     updated_resource = await update_status_and_publish_event_airlock_request(airlock_request, airlock_request_repo, user, AirlockRequestStatus.Submitted)
     return AirlockRequestInResponse(airlock_request=updated_resource)
+
+
+@airlock_workspace_router.post("/workspaces/{workspace_id}/requests/{airlock_request_id}/reviews", status_code=status.HTTP_200_OK, response_model=AirlockReviewInResponse, name=strings.API_REVIEW_AIRLOCK_REQUEST, dependencies=[Depends(get_workspace_by_id_from_path)])
+async def create_airlock_review(airlock_review_input: AirlockReviewInCreate, airlock_request=Depends(get_airlock_request_by_id_from_path), user=Depends(get_current_workspace_owner_or_researcher_user), airlock_request_repo=Depends(get_repository(AirlockRequestRepository)), airlock_review_repo=Depends(get_repository(AirlockReviewRepository)), workspace=Depends(get_deployed_workspace_by_id_from_path)) -> AirlockReviewInResponse:
```
[the remark this answers is not in the document]
Comment (eladiw) [reply api_app/api/routes/airlock.py:43]: correct, I think it should be only the ws owner

File: api_app/api/routes/airlock.py [hunk 3]
```diff
@@ -35,3 +37,17 @@ async def create_draft_request(airlock_request_input: AirlockRequestInCreate, us
...
+    # Create the review model and save in cosmos
+    try:
+        airlock_review = airlock_review_repo.create_airlock_review_item(airlock_review_input, workspace.id, airlock_request.id)
+    except (ValidationError, ValueError) as e:
+        logging.error(f"Failed create airlock review model instance: {e}")
```
Comment (eladiw) [inline api_app/api/routes/airlock.py:48]:
```suggestion
        logging.error(f"Failed creating airlock review model instance: {e}")
```

File: api_app/api/routes/airlock.py [hunk 4]
```diff
@@ -35,3 +37,17 @@ async def create_draft_request(airlock_request_input: AirlockRequestInCreate, us
...
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))
+    # Update the airlock request in cosmos, send a status_changed event
+    await update_status_and_publish_event_airlock_request(airlock_request, airlock_request_repo, user, airlock_review.reviewDecision)
+    await save_airlock_review(airlock_review, airlock_review_repo, user)
```
Comment (eladiw) [inline api_app/api/routes/airlock.py:52]: dont you want to first save the review.
imagine the saving of the review object fails, so in the current code, you already continued the flow, and the review is not saved == cant trace back.

prefer to save the review first, worst case the flow fails later, but it didnt continue until the trigger was saved

wdyt

File: api_app/event_grid/helpers.py [hunk 5]
```diff
@@ -28,7 +28,7 @@ async def send_status_changed_event(airlock_request: AirlockRequest):
     request_id = airlock_request.id
     status = airlock_request.status
     request_type = airlock_request.requestType
-    workspace_id = airlock_request.workspaceId
+    workspace_id = airlock_request.workspaceId[-4:]
```
Comment (eladiw) [inline api_app/event_grid/helpers.py:31]:
```suggestion
    short_workspace_id = airlock_request.workspaceId[-4:]
```

**Review by eladiw — CHANGES_REQUESTED** [review]
some minor changes

File: api_app/api/routes/airlock.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (anatbal) [reply api_app/api/routes/airlock.py]: Fixed

File: api_app/api/routes/airlock.py (hunk 1 above)
Comment (anatbal) [reply api_app/api/routes/airlock.py]: Fixed

File: api_app/api/routes/airlock.py (hunk 3 above)
Comment (anatbal) [reply api_app/api/routes/airlock.py]: Fixed

File: api_app/event_grid/helpers.py (hunk 5 above)
Comment (anatbal) [reply api_app/event_grid/helpers.py]: Fixed

**Pull request merged**